src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -29,294 +30,56 @@ internal class SilentRequest : RequestBase
-        }
-
-        private async Task<MsalTokenResponse> TryGetTokenUsingFociAsync(CancellationToken cancellationToken)
-        {
-            if (!ServiceBundle.PlatformProxy.GetFeatureFlags().IsFociEnabled)
+            catch(MsalException ex)
             {
-                return null;
-            }
-
-            var logger = AuthenticationRequestParameters.RequestContext.Logger;
-
-            // If the app was just added to the family, the app metadata will reflect this
-            // after the first RT exchanged.
-            bool? isFamilyMember = await CacheManager.IsAppFociMemberAsync(TheOnlyFamilyId).ConfigureAwait(false);
-
-            if (isFamilyMember.HasValue && !isFamilyMember.Value)
-            {
-                AuthenticationRequestParameters.RequestContext.Logger.Verbose(
-                    "[FOCI] App is not part of the family, skipping FOCI.");
-
-                return null;
-            }
-
-            logger.Verbose("[FOCI] App is part of the family or unknown, looking for FRT");
-            var familyRefreshToken = await CacheManager.FindFamilyRefreshTokenAsync(TheOnlyFamilyId).ConfigureAwait(false);
-            logger.Verbose("[FOCI] FRT found? " + (familyRefreshToken != null));
-
-            if (familyRefreshToken != null)
-            {
-                try
+                if (ex is MsalUiRequiredException || ex is MsalClientException)
                 {
-                    MsalTokenResponse frtTokenResponse = await RefreshAccessTokenAsync(familyRefreshToken, cancellationToken)
-                        .ConfigureAwait(false);
+                    var errorCode = ex.ErrorCode;
 
-                    logger.Verbose("[FOCI] FRT refresh succeeded");
-                    return frtTokenResponse;
-                }
-                catch (MsalServiceException ex)
-                {
-                    // Hack: STS does not yet send back the suberror on these platforms because they are not in an allowed list,
-                    // so the best thing we can do is to consider all errors as client_mismatch.
-#if NETSTANDARD || WINDOWS_APP || MAC
-                    ex?.GetType();  // avoid the "variable 'ex' is declared but never used" in this code path.
-                    return null;
-#else
-                    if (MsalError.InvalidGrantError.Equals(ex?.ErrorCode, StringComparison.OrdinalIgnoreCase) &&
-                        MsalError.ClientMismatch.Equals(ex?.SubError, StringComparison.OrdinalIgnoreCase))
+                    if (errorCode == MsalError.InvalidGrantError
+                     || errorCode == MsalError.NoTokensFoundError
+                     || errorCode == MsalError.NoAccountForLoginHint)
                     {
-                        logger.Error("[FOCI] FRT refresh failed - client mismatch");
-                        return null;
+                        _logger.Info("client auth strategy failed to acquire a token. Attempting to use broker strategy");
```

Comment:
>client [](start = 38, length = 6)

same comment as above. Maybe start the logger with the name of the class instead, or like bogdan did w/the PR on logging, where the name of the class is automatically included. then you'll know where we are in the code.